src/compiler/parser.rs

```diff
@@ -355,13 +356,13 @@ impl ParserType {
 
     fn parse_error_at_current_token(&self, message: &DiagnosticMessage) {
         self.parse_error_at(
-            self.scanner().get_token_pos(),
-            self.scanner().get_text_pos(),
+            self.scanner().get_token_pos().try_into().unwrap(),
```

Comment:
The scanner state stayed as `usize`'s (rather than also converting to `isize`'s) because they get used to index into the source text (which wants to be a `usize`), so using `.try_into().unwrap()` to convert `usize` -> `isize`